### PR TITLE
[CALCITE-2877] Make GeodeSchema constructor public to pass client cache object

### PR DESCRIPTION
**Pull request by chadasa**
[CALCITE-2877] Make GeodeSchema constructor public to pass client cache object

### CODE REVIEW

**Comment by asereda-gs** [comment]
Either this or #1022 should be merged

File: geode/src/main/java/org/apache/calcite/adapter/geode/rel/GeodeSchema.java [hunk 1]
```diff
@@ -41,13 +41,13 @@
   private final List<String> regionNames;
   private ImmutableMap<String, Table> tableMap;
 
-  GeodeSchema(String locatorHost, int locatorPort,
+  public GeodeSchema(String locatorHost, int locatorPort,
```
Comment (asereda-gs) [inline geode/src/main/java/org/apache/calcite/adapter/geode/rel/GeodeSchema.java]: If there is already a public constructor with `GemFireCache`  why do we need another with `locatorHost /  locatorPort` ? Is it possible to have just one constructor ?

**Comment by asereda-gs** [comment]
Overall LGTM. The only thing I'm debating is what arguments to provide to constructor:
1. `GemfireCache cache, Iterable<String> names` - current version
2. `GemfireCache cache, Predicate<String> filter`
3. `Iterable<Region<?,?>> regions`

**Comment by julianhyde** [comment]
The Schema should have no more functionality than the SchemaFactory - it is just allowed to accept arguments in "hydrated" form. Unless you have a plan for how to represent a Predicate as a string, you are straying away from that principle.

**Comment by chadasa** [comment]
I need to make one change inside GeodeUtils.java createRegion function to handle RegionExistsException.

**Comment by asereda-gs** [comment]
Thank you!

**Pull request merged**